### PR TITLE
Reduce package size

### PR DESCRIPTION
**Pull request by scotttrinh**
Avoids bundling large files like the changelog into the distributed package.

Unpacked size: 45.5 kB -> 12.3 kB
Packaged size: 15.7 kB -> 5 kB

Before:

```
npm pack --dry-run
npm notice
npm notice 📦  shell-quote@1.8.1
npm notice === Tarball Contents ===
npm notice 493B   .eslintrc
npm notice 582B   .github/FUNDING.yml
npm notice 229B   .nycrc
npm notice 21.7kB CHANGELOG.md
npm notice 1.1kB  LICENSE
npm notice 3.6kB  README.md
npm notice 128B   example/env.js
npm notice 106B   example/op.js
npm notice 118B   example/parse.js
npm notice 109B   example/quote.js
npm notice 87B    index.js
npm notice 1.8kB  package.json
npm notice 5.2kB  parse.js
npm notice 457B   quote.js
npm notice 295B   security.md
npm notice 642B   test/comment.js
npm notice 483B   test/env_fn.js
npm notice 1.9kB  test/env.js
npm notice 3.1kB  test/op.js
npm notice 1.4kB  test/parse.js
npm notice 1.4kB  test/quote.js
npm notice 565B   test/set.js
npm notice === Tarball Details ===
npm notice name:          shell-quote
npm notice version:       1.8.1
npm notice filename:      shell-quote-1.8.1.tgz
npm notice package size:  15.7 kB
npm notice unpacked size: 45.4 kB
npm notice shasum:        1e24f9b8fdc41b7c006d3efa6a94a0fa37f84a71
npm notice integrity:     sha512-ahgRvFkK1QU7C[...]k7EfVOHB4N27g==
npm notice total files:   22
npm notice
shell-quote-1.8.1.tgz
```

After:

```
npm pack --dry-run
npm notice
npm notice 📦  shell-quote@1.8.1
npm notice === Tarball Contents ===
npm notice 1.1kB LICENSE
npm notice 3.6kB README.md
npm notice 87B   index.js
npm notice 1.8kB package.json
npm notice 5.2kB parse.js
npm notice 457B  quote.js
npm notice === Tarball Details ===
npm notice name:          shell-quote
npm notice version:       1.8.1
npm notice filename:      shell-quote-1.8.1.tgz
npm notice package size:  5.0 kB
npm notice unpacked size: 12.3 kB
npm notice shasum:        452f8420a04b0105f9ac07200a8a35270ce13802
npm notice integrity:     sha512-5OlDuabL6IX/F[...]MoW/nSy7BNO7g==
npm notice total files:   6
npm notice
shell-quote-1.8.1.tgz
```

After including `tests` back in the package contents:

```
npm notice
npm notice 📦  shell-quote@1.8.1
npm notice === Tarball Contents ===
npm notice 582B  .github/FUNDING.yml
npm notice 1.1kB LICENSE
npm notice 3.6kB README.md
npm notice 87B   index.js
npm notice 1.8kB package.json
npm notice 5.2kB parse.js
npm notice 457B  quote.js
npm notice 295B  security.md
npm notice 642B  test/comment.js
npm notice 483B  test/env_fn.js
npm notice 1.9kB test/env.js
npm notice 3.1kB test/op.js
npm notice 1.4kB test/parse.js
npm notice 1.4kB test/quote.js
npm notice 565B  test/set.js
npm notice === Tarball Details ===
npm notice name:          shell-quote                        
npm notice version:       1.8.1                              
npm notice filename:      shell-quote-1.8.1.tgz              
npm notice package size:  7.7 kB                             
npm notice unpacked size: 22.6 kB                            
npm notice shasum:        10ffd15b781fa40af0826200d9114e63962db220
npm notice integrity:     sha512-9patS5xjm4tgs[...]ZK2kBuKah+e7g==
npm notice total files:   15                                 
npm notice
shell-quote-1.8.1.tgz
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,11 @@
 	],
 	"license": "MIT",
 	"main": "index.js",
+	"files": [
```
Comment (ljharb) [inline package.json:36]: The files field is dangerous and should NEVER be used; plus, we already use npmignore. If you want files ignored, add them to the publishConfig.ignore list below.

**Review by ljharb — CHANGES_REQUESTED** [review]
As written, this excludes way more than just the changelog - tests must always be included in the published package.

File: package.json (hunk 1 above)
Comment (scotttrinh) [reply package.json:36]: > The files field is dangerous and should NEVER be used;

TIL! Any resources to read about this? Couldn't find anything with a quick Google search.

**Comment by scotttrinh** [comment]
> tests must always be included in the published package.

While this is a perfectly fine stance to hold (your package after all!), I'm not sure I understand why a published package should include tests since consumers wouldn't be running tests for packages (most test dependencies would be dev dependencies, no?).

> If you want files ignored, add them to the publishConfig.ignore list below.

Yeah, I considered that, but I thought it was better to allow-list than deny-list certain files. This package is used by a lot of other packages, so reducing the package size of this one package down will reduce packaging size and distribution bandwidth of the overall npm ecosystem a lot, so seems worth having the smallest reasonable package size.

I'll update to liberally use `.npmignore` and let me know your feedback on which files you absolutely think need to be included in the 14 million weekly downloads this package has 😅

**Comment by scotttrinh** [comment]
Including the `tests` in the package contents makes the change less dramatic, but still an overall win at 50% savings:

Unpacked size: 45.5 kB -> 12.3 kB -> 22.6 kB
Packaged size: 15.7 kB -> 5 kB -> 7.7 kB

**Comment by ljharb** [comment]
Also, it’s not package authors’ responsibility to manage package size and distribution bandwidth, not maintainers’, and npm has chosen not to consider this a priority.

File: package.json [hunk 2]
```diff
@@ -61,7 +61,11 @@
 	},
 	"publishConfig": {
 		"ignore": [
-			".github/workflows"
+			".github/workflows",
+			"example",
+			".eslintrc",
```
Comment (ljharb) [inline package.json:66]: this will cause tests to fail
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -61,7 +61,11 @@
 	},
 	"publishConfig": {
 		"ignore": [
-			".github/workflows"
+			".github/workflows",
+			"example",
```
Comment (ljharb) [inline package.json:65]: while this is technically a breaking change, i doubt anyone’s relying on example files being present.

File: package.json [hunk 4]
```diff
@@ -61,7 +61,11 @@
...
+			".eslintrc",
+			".nycrc",
```
Comment (ljharb) [inline package.json:67]: i forget whether this file is required for tests to pass; you can check that locally by temporarily deleting it.

**Comment by scotttrinh** [comment]
> Also, it’s not package authors’ responsibility to manage package size and distribution bandwidth, it’s npm’s, and npm has chosen not to consider this a priority (by implementing separate “full” vs “minimal” tarballs for a single package)

Man, I'm learning so much here: I can't find any reference about this anywhere either (my Google-fu is weak tonight...) and when I install `shell-quote` in my own environment, I get all of the included files.

Maybe the minimal change that we can both agree to here is excluding the changelog? But then again, if it's not the package author's responsibility, we can also just close this PR as won't fix, no hard feelings. 🤝

**Comment by scotttrinh** [comment]
Latest changes:

Unpacked size: 45.5 kB -> 12.3 kB -> 22.6 kB -> 23.3 kB
Packaged size: 15.7 kB -> 5 kB -> 7.7 kB -> 8.0 kB

File: package.json (hunk 4 above)
Comment (ljharb) [reply package.json]: Right - the command is `npm explore foo && npm install && npm test`, which can be installed from a local cache or registry.

**Review by ljharb — APPROVED** [review]
I’m fine with this change, thanks.

**Pull request merged**